src/exposition/http.rs

```diff
@@ -270,65 +261,11 @@ mod handlers {
         Ok(content)
     }
 
-    pub async fn binary_metadata() -> Result<impl warp::Reply, Infallible> {
-        let mut metadata = Metadata::new();
-
-        // TODO(bmartin): this filtering needs to be consistent between the two
-        // binary output functions. This should be factored out to guarantee
-        // consistency.
-
-        // iterate through the static metrics and build-up the metadata object
-        // NOTE: dynamic metrics are omitted from the binary outputs, this is to
-        // ensure that the iteration order is consistent through the lifetime of
-        // the process.
-        for metric in metriken::metrics().static_metrics() {
-            let any = match metric.as_any() {
-                Some(any) => any,
-                None => {
-                    continue;
-                }
-            };
-
-            if metric.name().starts_with("log_") {
-                continue;
-            }
-
-            if let Some(_counter) = any.downcast_ref::<Counter>() {
-                metadata.counters.push(MetricMetadata {
-                    name: metric.formatted(metriken::Format::Simple),
-                });
-            } else if let Some(_gauge) = any.downcast_ref::<Gauge>() {
-                metadata.gauges.push(MetricMetadata {
-                    name: metric.formatted(metriken::Format::Simple),
-                });
-            } else if let Some(histogram) = any.downcast_ref::<RwLockHistogram>() {
-                metadata.histograms.push((
-                    MetricMetadata {
-                        name: metric.formatted(metriken::Format::Simple),
-                    },
-                    histogram.config(),
-                ));
-            }
-        }
-
-        let mut buf = Vec::new();
-        metadata.serialize(&mut Serializer::new(&mut buf)).unwrap();
-
-        Ok(buf)
-    }
-
-    pub async fn binary_readings() -> Result<impl warp::Reply, Infallible> {
-        let mut readings = Readings::new();
-
-        // TODO(bmartin): this filtering needs to be consistent between the two
-        // binary output functions. This should be factored out to guarantee
-        // consistency.
+    pub async fn msgpack() -> Result<impl warp::Reply, Infallible> {
+        let mut readings = MetricsSnapshot::new();
 
         // iterate through the static metrics and build-up the readings object
-        // NOTE: dynamic metrics are omitted from the binary outputs, this is to
-        // ensure that the iteration order is consistent through the lifetime of
-        // the process.
-        for metric in metriken::metrics().static_metrics() {
+        for metric in &metriken::metrics() {
```

Comment:
Nit: I guess comment should just say metrics, not static metrics.

Reply:
Fixed.